extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -200,8 +211,9 @@ private Table readTableInternal(
 
         if (partitionSpec.isUnpartitioned()) {
             // Create the flat layout location key finder
-            keyFinder = new IcebergFlatLayout(table, snapshot, fileIO, instructions);
+            keyFinder = new IcebergFlatLayout(tableDef, table, snapshot, fileIO, instructions);
         } else {
+            // Get the partitioning columns (applying column renames).
```

Comment:
If we agree that it's OK for the user to tell us to ignore some partitioning columns, this validation should come out and we should do _one_ of the following:
1. Drive the list of partitioning columns to look for, and their order, from the schema (in the key finder). The table code will just ignore the extra partitioning columns in the TLK. It might be weird that the TLK sorting takes *invisible* partitions into account.
2. Be sure to ignore extra schema columns in the key finder.